File: backend/app/api/routes/notifications.py
from typing import List

from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.models.models import Notification, User
from app.schemas.notification import NotificationOut
from app.core.deps import get_current_user

router = APIRouter(prefix="/notifications", tags=["notifications"])


@router.get("", response_model=List[NotificationOut])
def list_notifications(
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return db.query(Notification).filter(Notification.user_id == current_user.id).order_by(
        Notification.created_at.desc()
    ).all()


@router.patch("/{notification_id}/read", response_model=NotificationOut)
def mark_read(
    notification_id: str,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    note = db.query(Notification).filter(
        Notification.id == notification_id, Notification.user_id == current_user.id
    ).first()
    if not note:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Notification not found")
    note.is_read = True
    db.commit()
    db.refresh(note)
    return note


@router.patch("/read-all", status_code=status.HTTP_200_OK)
def mark_all_read(
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    db.query(Notification).filter(
        Notification.user_id == current_user.id, Notification.is_read.is_(False)
    ).update({"is_read": True})
    db.commit()
    return {"success": True}

File: frontend/src/app/services/notifications.ts
import { api } from "./api";

export interface Notification {
  id: string;
  user_id: string;
  type: "APPOINTMENT" | "MESSAGE" | "PRESCRIPTION" | "SYSTEM" | "REMINDER";
  title: string;
  body: string;
  is_read: boolean;
  meta?: any;
  created_at: string;
}

export const notificationsService = {
  async getMine(): Promise<Notification[]> {
    return api.get("/notifications");
  },

  async markRead(id: string): Promise<Notification> {
    return api.patch(`/notifications/${id}/read`);
  },

  async markAllRead(): Promise<any> {
    return api.patch("/notifications/read-all");
  }
};
